circus-train-avro/src/main/java/com/hotels/bdp/circustrain/avro/transformation/AbstractAvroSerDeTransformation.java

```diff
@@ -50,6 +55,28 @@ protected boolean avroTransformationSpecified() {
     return argsPresent(getAvroSchemaDestinationFolder(), getEventId());
   }
 
+  protected String constructSource(String source) {
+    String scheme = null;
+    Object overrideScheme = avroSerdeConfigOverride
+        .get(AvroSerDeConfig.TABLE_REPLICATION_OVERRIDE_SOURCE_AVRO_URL_SCHEME);
+    if (overrideScheme != null && StringUtils.isNotBlank(overrideScheme.toString())) {
+      scheme = overrideScheme.toString();
+    } else {
+      scheme = avroSerDeConfig.getSourceAvroUrlScheme();
+    }
+
+    if (isNotEmpty(scheme)) {
+      try {
+        URI uri = new URI(source);
+        source = new URI(scheme, uri.getUserInfo(), uri.getHost(), uri.getPort(), uri.getPath(), uri.getQuery(),
```

Comment:
Would all this work if instead of doing all this you just prefixed "source" with the passed in scheme and did nothing more? i.e. none of this URI stuff, you just append the two strings and pass it back out?

Reply:
Lots of edge cases, this is better imo